### PR TITLE
Calculate total transaction count + gas usage at db level

### PR DESCRIPTION
**Pull request by rkachowski**
_[GitHub keywords to close any associated issues](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)_

### Description

Calculates tx count + gas usage via db trigger mechanism. Upon each modification to the `transactions` table, a secondary table of aggregate stats will be updated with information about the state of the transactions. 

> Note: the attached migration will require a scaling down of the indexer to get an accurate count of the transactions. This migration should take around 30 minutes to complete (based on GCP query insights timing)
 
The method in `Explorer.Chain` keeps the name `transaction_estimated_count()` for the sake of compatibility with the API throughout the application and with the upstream code base. 

 ### Other changes

* Removed existing `TransactionCache` and `GasUsage` modules

### Tested

* Ran migration on `blockscoutstagingrc1` and asserted behavior of db during insertions

### Issues

 - Fixes celo-org/data-services#236

### CODE REVIEW

File: apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+defmodule Explorer.Repo.Migrations.SetCountersOnTransactionTable do
+  use Ecto.Migration
+
+  def up do
+    execute("CREATE TABLE celo_transaction_stats(stat_type varchar(255), value bigint);")
```
Comment (carterqw2) [inline apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs:5]: `bigint` looks too small for the total gas used value, no? I'd go with `numeric(100,0)` like in other similar cases.

File: apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      UPDATE celo_transaction_stats SET value = 0 WHERE stat_type = 'total_gas_used';
+
+      RETURN NULL;
+    END IF;
+    END;$$;
+    """)
+
+    execute("""
+    CREATE CONSTRAINT TRIGGER celo_transaction_stats_modified
+    AFTER INSERT OR DELETE ON transactions
```
Comment (carterqw2) [inline apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs:37]: `UPDATE` missing?

**Review by carterqw2 — CHANGES_REQUESTED** [review]

File: apps/explorer/lib/explorer/chain/celo/transaction_stats.ex [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  alias Explorer.Repo
+  use Explorer.Schema
+  import Ecto.Query
+
+  @type t :: %__MODULE__{
+          stat_type: String.t(),
+          value: non_neg_integer()
+        }
+
+  @attrs ~w( stat_type value)a
```
[the remark this answers is not in the document]
Comment (rkachowski) [reply apps/explorer/lib/explorer/chain/celo/transaction_stats.ex:17]: yeah it is, I'm not sure what happened there

File: apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs (hunk 2 above)
Comment (rkachowski) [reply apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs:37]: whoa, nice catch

File: apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      UPDATE celo_transaction_stats SET value = 0 WHERE stat_type = 'total_transaction_count';
+      UPDATE celo_transaction_stats SET value = 0 WHERE stat_type = 'total_gas_used';
+
+      RETURN NULL;
+    END IF;
+    END;$$;
+    """)
+
+    execute("""
+    CREATE CONSTRAINT TRIGGER celo_transaction_stats_modified
```
Comment (carterqw2) [inline apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs:36]: Ah, sorry, missed `OR REPLACE` here and below.

**Review by carterqw2 — APPROVED** [review]

File: apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs (hunk 4 above)
Comment (rkachowski) [reply apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs]: https://github.com/celo-org/blockscout/runs/7004593995?check_suite_focus=true#step:10:1593

```
13:07:52.744 [info]  execute "CREATE OR REPLACE CONSTRAINT TRIGGER celo_transaction_stats_modified\nAFTER INSERT OR DELETE OR UPDATE ON transactions\nDEFERRABLE INITIALLY DEFERRED\nFOR EACH ROW EXECUTE PROCEDURE celo_transaction_stats_trigger_func();\n"

** (Postgrex.Error) ERROR 0A000 (feature_not_supported) CREATE OR REPLACE CONSTRAINT TRIGGER is not supported
```

It seems like this syntax doesn't exist for triggers

Comment (rkachowski) [reply apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs]: https://stackoverflow.com/questions/35927365/create-or-replace-trigger-postgres#comment77864054_39650162

Comment (carterqw2) [reply apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs:36]: https://www.postgresql.org/docs/current/sql-createtrigger.html it's not supported by our postgres version it seems

**Pull request merged**